### PR TITLE
Add unit tests for `coriolis/taskflow` modules 

### PR DESCRIPTION
**Pull request by mihaelabalutoiu**
This PR adds unit tests for the following modules:
* `taskflow/runner.py`
* `taskflow/utils.py`
* `taskflow/base.py`

### CODE REVIEW

**Review by Dany9966 — CHANGES_REQUESTED** [review]

File: coriolis/tests/taskflow/test_base.py [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+        self.flow_failures = mock.Mock()
+        self.mock_instance = mock.Mock()
+
+    def test_get_error_str_for_flow_failures(self):
+        self.task_failure.exception_str = "Exception"
+        self.task_failure.traceback_str = "Traceback"
+
+        result = base.BaseCoriolisTaskflowTask.\
+            _get_error_str_for_flow_failures(None, {"task1":
+                                                    self.task_failure})
```
[the remark this answers is not in the document]
Comment (mihaelabalutoiu) [reply coriolis/tests/taskflow/test_base.py:41]: Gotcha! Thank you!

File: coriolis/tests/taskflow/test_base.py [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+        self.assertEqual(kwargs, {'provides':
+                                  ['test', 'task-test_task-result']})
+
+    def test_set_requires_for_dependencies(self):
+        self.task._set_requires_for_dependencies(self.kwargs, self.depends_on)
+        self.assertEqual(self.kwargs,
+                         {'test_task_id': 'test_task_id',
+                          'requires': ['task-test_task_id-result']})
+
+    def test_set_requires_for_dependencies_with_provides_key(self):
```
[the remark this answers is not in the document]
Comment (mihaelabalutoiu) [reply coriolis/tests/taskflow/test_base.py:127]: Yes, you are right! Thanks!

File: coriolis/tests/taskflow/test_base.py [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+        kwargs = {'requires': ['test']}
+        self.task._set_requires_for_dependencies(kwargs, self.depends_on)
+        self.assertEqual(kwargs,
+                         {'requires': ['test', 'task-test_task_id-result']})
+
+    @mock.patch('coriolis.tasks.factory.get_task_runner_class')
+    def test_set_requires_for_task_info_fields(self, mock_get_task_runner):
+        self.mock_task_runner.get_required_task_info_properties.\
+            return_value = ['test']
```
[the remark this answers is not in the document]
Comment (mihaelabalutoiu) [reply coriolis/tests/taskflow/test_base.py:136]: Good idea!

**Comment by mihaelabalutoiu** [comment]
Done. I made the changes. Please take a look now, thanks.

**Review by Dany9966 — APPROVED** [review]
LGTM

**Pull request merged**